be/src/storage/tablet_meta_manager.cpp

```diff
@@ -803,7 +803,8 @@ Status TabletMetaManager::rowset_iterate(DataDir* store, TTabletId tablet_id, co
 
 Status TabletMetaManager::apply_rowset_commit(DataDir* store, TTabletId tablet_id, int64_t logid,
                                               const EditVersion& version,
-                                              vector<std::pair<uint32_t, DelVectorPtr>>& delvecs) {
+                                              vector<std::pair<uint32_t, DelVectorPtr>>& delvecs,
+                                              const PersistentIndexMetaPB& index_meta) {
```

Comment:
What if enable_persistent_index is false, is there still metapb in meta?

Reply:
> What if enable_persistent_index is false, is there still metapb in meta?

done, change in https://github.com/StarRocks/starrocks/pull/4174/commits/2cf69fa2ab845233972d29ad12997c014d8b9691

If `enable_persistent_index` is false, don't update metapb in TabletMeta